src/uct/ib/dc/dc_mlx5.c

```diff
@@ -913,21 +913,32 @@ void uct_dc_mlx5_iface_init_version(uct_dc_mlx5_iface_t *iface, uct_md_h md)
     }
 }
 
-int uct_dc_mlx5_iface_is_reachable(const uct_iface_h tl_iface,
-                                   const uct_device_addr_t *dev_addr,
-                                   const uct_iface_addr_t *iface_addr)
+static int
+uct_dc_mlx5_iface_is_reachable_v2(const uct_iface_h tl_iface,
+                                  const uct_iface_is_reachable_params_t *params)
 {
-    uct_dc_mlx5_iface_addr_t *addr = (uct_dc_mlx5_iface_addr_t *)iface_addr;
-    uct_dc_mlx5_iface_t *iface;
+    uct_dc_mlx5_iface_t *iface = ucs_derived_of(tl_iface, uct_dc_mlx5_iface_t);
+    const uct_dc_mlx5_iface_addr_t *addr;
+    int same_tm, same_version;
 
-    iface = ucs_derived_of(tl_iface, uct_dc_mlx5_iface_t);
-    ucs_assert_always(iface_addr != NULL);
+    if (!uct_iface_is_reachable_params_valid(
+                params, UCT_IFACE_IS_REACHABLE_FIELD_DEVICE_ADDR)) {
+        return 0;
+    }
+
```

Comment:
this is already checked in `uct_ib_iface_is_reachable_v2`
you do not check it in `uct_rc_mlx5_iface_is_reachable_v2` which seems to be the right thing to do